pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -1643,6 +1643,32 @@ func TestHttpProxyListener(t *testing.T) {
 	}
 }
 
+func TestHttpProxyListenerPerWorkload(t *testing.T) {
+	p := &fakePlugin{}
+	configgen := NewConfigGenerator([]plugin.Plugin{p}, &model.DisabledCache{})
```

Comment:
Can you please add a test case to cover the case that has mesh value and show that sidecar one overrides it as well - just for completeness?

Reply:
The intent was to eventually deprecate the MeshConfig value - in reality both tests are pretty superficial. I will add more testing - I am working on some environments without iptables. It's not a big deal to cut&paste this test again and set both, but not sure it adds much value.

Reply:
If it is permanent it makes sense to add test. If it is temporary and mesh config value is deprecated, I am fine without test.

Reply:
Yes, mesh config HttpProxyPort needs to be deprecated - it's not really usable and doesn't make sense. You may have few machines or docker images where iptables is missing - but setting the global flag will increase the config size for ALL  the other workloads in the mesh.

Technically the entire MeshConfig needs to be deprecated and replaced with narrower-scoped APIs, we're starting with ProxyConfig.